src/Kiota.Builder/Refiners/JavaRefiner.cs

```diff
@@ -155,24 +155,24 @@ public override void Refine(CodeNamespace generatedCode)
                 currentMethod.Parameters.Where(x => x.IsOfKind(CodeParameterKind.HttpCore, CodeParameterKind.CurrentPath)).ToList().ForEach(x => x.Type.IsNullable = true);
                 currentMethod.Parameters.Where(x => x.IsOfKind(CodeParameterKind.HttpCore) && x.Type.Name.StartsWith("i", StringComparison.OrdinalIgnoreCase)).ToList().ForEach(x => x.Type.Name = x.Type.Name[1..]); // removing the "I" 
             }
-        }
-        private static void AddRequireNonNullImports(CodeElement currentElement) {
-            if(currentElement is CodeMethod currentMethod && currentMethod.Parameters.Any(x => !x.Optional)) {
-                var parentClass = currentMethod.Parent as CodeClass;
-                var newUsing = new CodeUsing(parentClass) {
-                    Name = "Objects",
-                };
-                newUsing.Declaration = new CodeType(newUsing) {
-                    Name = "java.util",
-                    IsExternal = true,
+            if (currentMethod.IsOfKind(CodeMethodKind.RequestBuilderWithParameters, CodeMethodKind.Constructor) &&
+                    currentMethod.Parameters.Any(x => OriginalDateTimeOffsetType.Equals(x.Type.Name, StringComparison.OrdinalIgnoreCase))) {
+                currentMethod.Parameters.Where(x => OriginalDateTimeOffsetType.Equals(x.Type.Name, StringComparison.OrdinalIgnoreCase))
+                                        .ToList()
+                                        .ForEach(x => x.Type.Name = JavaOffsetDateTimeType);
+                var nUsing = new CodeUsing {
+                    Name = JavaOffsetDateTimeType,
+                    Declaration = new CodeType {
+                        Name = JavaOffsetDateTimeTypePackage,
+                        IsExternal = true,
+                    },
                 };
-                parentClass?.AddUsing(newUsing);
+                (currentMethod.Parent as CodeClass).AddUsing(nUsing);
```

Comment:
```suggestion
                (currentMethod.Parent as CodeClass)?.AddUsing(nUsing);
```

Reply:
thanks, fixed using type matching in the condition